api/v3/openapi.yaml

```diff
@@ -2271,6 +2302,69 @@ components:
               items:
                 $ref: '#/components/schemas/PartitionData'
 
+    ProduceBatchErrorEntry:
+      type: object
+      required:
+        - id
+        - error_code
+      properties:
+        id:
+          description: Batch entry ID.
+          type: string
+          minLength: 1
+          maxLength: 80
+        error_code:
+          description: Error code.
+          type: integer
+          format: int32
+        message:
+          description: Error message.
+          type: string
+
+    ProduceBatchResponse:
+      description: Response for producing a batch of records.
+      type: object
+      properties:
+        successes:
+          description: Successful batch results.
+          type: array
+          items:
+            $ref: '#/components/schemas/ProduceBatchResultEntry'
+        failures:
+          description: Failed batch results.
+          type: array
+          items:
+            $ref: '#/components/schemas/ProduceBatchErrorEntry'
```

Comment:
nit: `ProduceBatchErrorEntry` -> `ProduceBatchFailureEntry`

for consistency, since the its part of the array "failures"